src/data_collectors/dynamics_data_collector.py

```diff
@@ -0,0 +1,32 @@
+from typing import Any
+
+import torch
+from torch import Tensor
+
+from ..utils.step_record import RecordKeys as RK
+from .data_collector import DataCollector
+
+
+class DynamicsDataCollector(DataCollector):
+    def __init__(self):
+        self.prev_actions = []
+        self.observations = []
+        self.actions = []
+        self.next_observations = []
+
+    def collect(self, step_record: dict[str, Any]):
+        self.prev_actions.append(step_record[RK.PREVIOUS_ACTION])
+        self.observations.append(step_record[RK.OBSERVATION])
+        self.actions.append(step_record[RK.ACTION])
+        self.next_observations.append(step_record[RK.NEXT_OBSERVATION])
+
+    def get_data(self) -> (Tensor, Tensor, Tensor, Tensor):
+        prev_actions = torch.stack(self.prev_actions)
+        self.prev_actions = []
+        observations = torch.stack(self.observations)
+        self.observations = []
+        actions = torch.stack(self.actions)
+        self.actions = []
+        next_observations = torch.stack(self.next_observations)
+        self.next_observations = []
```

Comment:
- InveseDynamicsのデータセットはリセットする必要がなく、AMIの活動全体を通して使い続けることができます。
```suggestion
    def get_data(self) -> (Tensor, Tensor, Tensor, Tensor):
        prev_actions = torch.stack(self.prev_actions)
        observations = torch.stack(self.observations)
        actions = torch.stack(self.actions)
        next_observations = torch.stack(self.next_observations)
```

Reply:
https://www.notion.so/911e4788afab4c0798406b31c51a2500?pvs=4#cd9888bf9eeb4b4a95ca5b9ca6ad4590
に従い確率的に消滅するようにしました https://github.com/MLShukai/PrimitiveAMI/pull/53/commits/4a4568ef921f17cc08960c2f27c99f3123475b05